src/escape.d

```diff
@@ -310,6 +310,23 @@ bool checkEscapeRef(Scope* sc, Expression e, bool gag)
                             tf.isreturn = true;
                         }
                     }
+                    else
+                    {
+                        // Set STCreturn in the Parameter storage class corresponding to v
+                        TypeFunction tf = cast(TypeFunction)sc.func.type;
+                        if (tf.parameters)
+                        {
+                            const dim = Parameter.dim(tf.parameters);
+                            foreach (const i; 0 .. dim)
+                            {
+                                Parameter p = Parameter.getNth(tf.parameters, i);
+                                if (p.ident == v.ident)
+                                {
+                                    p.storageClass |= STCreturn;
```

Comment:
You won't, because the identifiers are all in the same name space, so must be unique.